cmd/agent/subcommands/jmx/command.go

```diff
@@ -138,6 +140,7 @@ func Commands(globalParams *command.GlobalParams) []*cobra.Command {
 			fx.Provide(func() demultiplexer.Component { return nil }),
 			fx.Provide(func() inventorychecks.Component { return nil }),
 			fx.Provide(func() packagesigning.Component { return nil }),
+			fx.Provide(func() optional.Option[rcservice.Component] { return optional.NewOption[rcservice.Component](nil) }),
```

Comment:
Note: If the component is never enabled, you can use instead `optional.NewNone[rcservice.Component]()`